### PR TITLE
Adding a new api to fetch the health details of the resource "vault". Apart form this this PR also contains some fixes based on  some new rules added in the auto rest linter tool.

### PR DESCRIPTION
**Pull request by mukum**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by dsgouda** [comment]
Sorry to bother, but reopening PR to test CI builds.

**Comment by veronicagg** [comment]
@ravbhatnagar please take a look, thanks!

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Since this is not a tracked resource, you dont need location in the response body

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: I think this could be an enum with a known set of values.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: I think this could be an enum with a known set of values.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: I think this could be an enum with a known set of values.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Is this API also giving the health summary for all the child resources of vault (fabric/ protectedItem etc)? If yes, then does the caller of this API need to have read access on all the underlying resources as well.  Also, from the API signature it is unclear. Moreover, in the payload it will be unclear which "replicationItem" does a particular issue correspond to? Am I missing something?

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: What is this? what could be the resource types? How do you use this?

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: What is this value and how will the customer use this?

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Could this be an enum

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Could this be an enum

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: What is this entityId?

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Which child error corresponds to which event? how will this be used?

Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: This is a new property getting added. Will need a api-version update.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: How will this model be used?

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (ravbhatnagar) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: So you the health status property has been added to the GETs of fabric etc. Why do you need a separate API to just get the healthStatus? Also, new properties have been added all across this swagger. This would require an api-version change on both service and swagger.

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]
@mukum - This PR contains a lot of changes like new property additions, newly added models, adding APIs etc. We only discussed the new API changes in our meeting and so Many of the other changes are difficult to understand. Once you have addressed the questions/comments in the PR, please schedule a review with the Azure API review board ( @johanste , @NiklasGustafsson ) and myself.

**Comment by ravbhatnagar** [comment]
@veronicagg - This may need to be reviewed in person with the Azure API review board. The newly added properties and models are not are not self explanatory. The API that is being added is fine and feedback has been provided on that.

**Comment by veronicagg** [comment]
@ravbhatnagar thanks! will wait for API review board sign off.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: No gaurav, This API will give give the health summary of immediate child resources only. At the vault level we are going with the exception which would give protected items level summary as well.
Yes the caller of the API will need to have the read access on the underlying resources and we will give the same. In the payload you can see each health summary contains the list of affectedResourceCorrelationId to related this health impacts which all resources.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Agree, But since all our proxy resource derive from a common base class hence it is there.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: No This won't be.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: This is the correlation id of the health with the impacted resource.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: These are pre approved classes/object model and any change now will be a breaking change.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: All errors relate to the same entity. There is a "root cause and impact" kind of relationship between them hence root cause is the parent and impact health errors are the child.

Since all are object are discovered object(not client created) hence this new property added  shouldn't require the api-version update.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: The health summary detail is the summary of the container level resource. The purpose of this is to give a dashboard kind of view at the container level resource like vault/fabric. As stated earlier since we have discovered resources only adding a new property shouldn't be a breaking change.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Since this is the error code only which we have exposed at other classes as well, like health error, we don't feel this to be exposed as an enum hence going ahead with string.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Agree changing it to enum.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Agree changing it to enum.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: This is our resource type defined for the proxy resources. This is to identify the resource type of the resources impacted by the certain category of health issue.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: This the the correlationId for the impacted resources, which is nothing but the arm id's of the resources for all the practical purpose.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: This will be used to query the list of events based on the filters provided as part of the EventQueryParameter. This will have value like startdate to enddate, severity of the filter... impacted resourcee name and so...on..

**Review by veronicagg — CHANGES_REQUESTED** [review]
Please review linter errors at https://travis-ci.org/Azure/azure-rest-api-specs/jobs/261175730#L675
Please review examples errors - invalid format for date-time: 
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/261175732#L706
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/261175732#L722
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/261175732#L775

**Comment by veronicagg** [comment]
@mukum @ravbhatnagar have you been able to sync on ARM feedback?

**Comment by ravbhatnagar** [comment]
@veronicagg - No, we haven't been able to agree on the right design for this API yet. We are close though and should be able to close today or Monday. @mukum - FYI.

**Comment by veronicagg** [comment]
@ravbhatnagar once you're good with the changes on this PR, please label it and let me know, thanks!

**Comment by ravbhatnagar** [comment]
@mukum - The change for the new API looks good now. However, as discussed earlier it seems that there are new properties in this swagger that were already added on the service side but getting added to swagger now. Please confirm if thats the case. Because in such cases, an api-version update is required. 
@veronicagg - SDK can start reviewing.

**Comment by veronicagg** [comment]
thanks @ravbhatnagar.
@mukum have you been able to look into the errors from CI that I pointed out before? Feel free to let me know if you have questions.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: Hi Veronica, yes it is a read only property. Our service is bit different as we have the get and put object schema different, hence even if we don't mark it read only the client cannot used the same object received in the get call to Put/Update. Although i agree marking a property read only is a good practice and we have this as one of our backlog item while moving to the next api version.

**Comment by mukum** [comment]
@veronicagg 
Hi Veronica, 
   I went through the errors from CL, all the errors are for camel casing of the property name(for the older set of properties name). We have this as one of the backlog items once we move to the newer api version.
One other is the Job object which has few properties other than Id type and Location at the top level. I talked with my colleague regarding the same and as per him we had got exception for this last time. Since we are not upgrading the version of the API we are not fixing it, will take it in the next iteration when we move to the newer api version.

**Comment by mukum** [comment]
@ravbhatnagar  as discussed earlier, since we have the get and put object schema completely different, hence adding a new property in the get object schema doesn't require a version update.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (veronicagg) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: ok, thanks, yes, I would recommend doing that in your next version if you don't see it being possible now (or causes a breaking change).

**Review by veronicagg — CHANGES_REQUESTED** [review]
Thanks @mukum . 
On camel casing, ok. 
For Job model which has additional properties at the top level, as far as I know it's not allowed for ARM, @ravbhatnagar could you confirm you approved this?
Have you considered flattening properties per the warnings at https://travis-ci.org/Azure/azure-rest-api-specs/jobs/264396941#L1286?
Have you evaluated if the boolean properties flagged should be considered for enums? ex. https://travis-ci.org/Azure/azure-rest-api-specs/jobs/264396941#L1934
Do you have any questions on the "TrackedResource...." warnings? have you evaluated them?
thanks!

**Comment by veronicagg** [comment]
@mukum there looks to be some model validation errors too, please review: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/264396943#L657

**Comment by veronicagg** [comment]
@ravbhatnagar please add the sign-off label once you approve this PR. I see that there are breaking changes being introduced, so I'd like have confirmation on sign-off from ARM team.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json
Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: @avrai @vijaynar Please add the missing examples if any.

Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: @veronicagg For Job model this is the PR"https://github.com/Azure/azure-rest-api-specs/pull/1209" where i guess it was closed. Since we are not changing the version of the API we are not making that change.

Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: @veronicagg Booleans properties we have evaluated and we don't feel the need to change it to enum.

Comment (mukum) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/2016-08-10/service.json]: @veronicagg Gone through the different tracked resources warning. Since there is no version upgrade we don't want to introduce new api's. We will try to accommodate the warning scenarios with the newer version release.

**Comment by ravbhatnagar** [comment]
@veronicagg - from what I can see in the PR, jobs is an existing model. So issues in the model are older issues. As discussed we will get issues with existing APIs/models fixed in a separate cycle. Other property additions have already happened in the service and these are being added to swagger now. 
@mukum - please note that the jobs model will need to be fixed since it contains some top level properties which are not allowed as per ARM RPC. These properties will need to be moved inside the properties bag in your next api-version rev.

**Review by ravbhatnagar — APPROVED** [review]

**Comment by veronicagg** [comment]
Thanks @ravbhatnagar. 
@mukum Please add the pending example and open & link the issue we talked about tracking the update of the examples that are failing (due to bug fixes in our tool). With those, I can approve. Thanks!

**Comment by veronicagg** [comment]
@mukum any updates?

**Comment by salameer** [comment]
@mukum please respond to the feedback to avoid  closing this PR due to lack of activity

Thanks

Samer

**Comment by veronicagg** [comment]
Thanks @mukum for addressing the issues. 
Samples are compliant, though it's good idea to consider making them specific, so that you get more coverage on the properties, you may want to add this to the issue you linked. 
Regarding the removed operation, if it's implemented on the service time, it may get reported as not in the swagger by the ARM team, just so that you're aware - FYI @ravbhatnagar

**Review by veronicagg — APPROVED** [review]

**Pull request merged**
